docs/public/_redirects

```diff
@@ -17,6 +17,7 @@
 /x/license/ https://mui.com/store/legal/mui-x-eula/ 301
 /x/react-data-grid/group-pivot/ /x/react-data-grid/tree-data/ 301
 /:lang/x/react-data-grid/group-pivot/ /:lang/x/react-data-grid/tree-data/ 301
+/mui.com/x/react-data-grid/columns/ /mui.com/x/react-data-grid/column-definition/ 301
```

Comment:
I think the redirects are not working. I would expect something like this to work though.

```suggestion
/x/react-data-grid/columns/#:feature /x/react-data-grid/:feature 301
```

Reply:
I think it's impossible to redirect based on the hash

Reply:
I don't know why I put the mui.com at the beginning (not enough coffee maybe ^^)

About the `#:feature` the problem is that hash is not supported by 301

In fact, everything after the `/#` is not send to the server, that's a nice trick when hosting single page application in github pages

Reply:
Alright. would we then need to create a redirect for each feature?

The current rule pushes for instance 
https://deploy-preview-4940--material-ui-x.netlify.app/x/react-data-grid/columns/#column-groups
to 
https://deploy-preview-4940--material-ui-x.netlify.app/x/react-data-grid/column-definition#column-groups

Reply:
I thought about it, but the problem is the server only  receives a request for  `/x/react-data-grid/columns/` so it has no way to know which sub-pages is the good one

Since the page does not contain id `column-groups` going to `/column-definition/#column-groups` will have the same behavior as `/column-definition/`

From the hash, we could decide to do a redirect directly in the javascript. But it will be complicated, and a nightmare to maintain

Reply:
No, we cannot redirect URLs that contain # - see https://github.com/mui/material-ui/pull/32504#discussion_r861989714

Reply:
Got it, the redirect here is only to avoid the 404.
I'll open a PR on the website updating the links there.

Reply:
ah, it's done already, nice!

Reply:
@alexfauquette already opened a PR here https://github.com/mui/material-ui/pull/32842
Or do you mean something else?